src/NexusMods.Paths/FileSystemAbstraction/IFileSystem.cs

```diff
@@ -0,0 +1,162 @@
+using JetBrains.Annotations;
+
+namespace NexusMods.Paths;
+
+/// <summary>
+/// File system abstraction.
+/// </summary>
+[PublicAPI]
+public interface IFileSystem
+{
+    /// <summary>
+    /// Creates a new <see cref="FileSystem"/> that allows for path mapping.
+    /// </summary>
+    /// <param name="pathMappings">Path mappings</param>
+    /// <returns></returns>
+    IFileSystem CreateOverlayFileSystem(Dictionary<AbsolutePath, AbsolutePath> pathMappings);
+
+    AbsolutePath FromFullPath(string fullPath);
```

Comment:
Missing XML docs here; and namespace doesn't match up.
If the latter is intended, suppress it; otherwise make it match up the directory structure [preferred].